include/nc.h

```diff
@@ -137,4 +137,12 @@ EXTERNL int NC_copy_data_all(NC* nc, nc_type xtypeid, const void* memory, size_t
 #define USED2INFO(nc) ((1<<(nc->dispatch->model)) & (1<<NC_FORMATX_DAP2))
 #define USED4INFO(nc) ((1<<(nc->dispatch->model)) & (1<<NC_FORMATX_DAP4))
 
+/* In DAP4 and Zarr (and maybe other places in the future)
+   we may have dimensions with a size, but no name.
+   In this case we need to create a name based on the size.
+   As a rule, the dimension name is NCDIMANON_<n> where n is the size
+   and NCDIMANON is a prefix defined here.
+*/
+#define NCDIMANON "_Anonymous_Dim"   
```

Comment:
Does this leave the door open for potential name collisions?

Reply:
I don't think so. If I have an anonymous dimension of, say, size 20, then the actual created
dimension will have the name _Anonymous_Dim_20. So there can be no conflict with 
anonymous dimensions of different size.
Further,  ** all**  anonymous dimensions are created in the root group. So
for example, all anonymous dimensions of size 20 will find and re-use the corresponding
dimension in the root group.